scrapy/pipelines/images.py

```diff
@@ -38,6 +38,15 @@ class ImagesPipeline(FilesPipeline):
 
     MEDIA_NAME = 'image'
 
+    _DEPRECATED_ATTRS = {
+        'IMAGES_EXPIRES': 'expires',
+        'IMAGES_URLS_FIELD': 'images_urls_field',
+        'IMAGES_RESULT_FIELD': 'images_result_field',
+        'IMAGES_MIN_WIDTH': 'min_width',
+        'IMAGES_MIN_HEIGHT': 'min_height',
+        'IMAGES_THUMBS': 'thumbs',
+    }
+
     def __init__(self, store_uri, download_func=None, settings=None):
```

Comment:
dont you think that perhaps easier approach would be just having uppercase class attributes and defaulting to them in pipeline init method? So dont add shim for getattr but rather change things in such a way that 

a) deprecated keys are not deprecated, but they are kept as class attributes. 
b) pipeline init checks for settings values, if there are not settings it defaults to class attributes
c) pipeline uses lowercase attributes all around

This would allow us to have cake and eat it, and has following benefits:

* there are no problems with backward compatibility
* if someone is like me and has multiple pipelines he can just define different class attributes in each one of them and thus have different settings for each pipeline.

Drawback of this solution is:
* we dont deprecate class attributes (but do we really need it?)

Perhaps there are other drawbacks I'm not aware of.